### ISSUE TITLE
matlab plug-in 

### ISSUE THREAD

**Issue by karenacai** [stevengj/nlopt#147]
When I run the following script in MATLAB, 
```
opt.algorithm = NLOPT_LD_MMA;
opt.lower_bounds = [-inf, 0];
opt.min_objective = @myfunc;
opt.fc_tol = [1e-8, 1e-8];
opt.xtol_rel = 1e-4;

[xopt, fmin, retcode] = nlopt_optimize(opt, [1.234 5.678]);
```
I get the MATLAB error: "undefined function or variable 'nlopt_optimize'.

I tried to do the following to debug the error: 

I followed the installation instructions at http://ab-initio.mit.edu/wiki/index.php/NLopt_Installation. I am not quite sure I did everything correctly. What is the "/path/to/mex" directory we are supposed to set the MEX variable to? 

Also, I found the file "nlopt_optimize-mex.c" in the subdirectory: nlopt-2.4.2/octave and tried to run the command: 

>mex -output nlopt_optimize nlopt_optimize-mex.c 

to compile the program but have been getting the following linking errors: 
```
/tmp/mex_35024843806807_3261/nlopt_optimize-mex.o: In function
`user_function':
nlopt_optimize-mex.c:(.text+0x4d5): undefined reference to
`nlopt_force_stop'
/tmp/mex_35024843806807_3261/nlopt_optimize-mex.o: In function
`make_opt':
nlopt_optimize-mex.c:(.text+0x532): undefined reference to
`nlopt_destroy'
nlopt_optimize-mex.c:(.text+0x53c): undefined reference to
`nlopt_destroy' 
...
nlopt_optimize-mex.c:(.text+0x11e0): undefined reference to
`nlopt_destroy'
nlopt_optimize-mex.c:(.text+0x1295): undefined reference to
`nlopt_add_equality_constraint'
nlopt_optimize-mex.c:(.text+0x12b9): undefined reference to
`nlopt_destroy'
nlopt_optimize-mex.c:(.text+0x1340): undefined reference to
`nlopt_optimize'
nlopt_optimize-mex.c:(.text+0x1392): undefined reference to
`nlopt_destroy'
collect2: error: ld returned 1 exit status
```
I found the same issue was addressed here: https://www.mail-archive.com/nlopt-discuss@ab-initio.mit.edu/msg00303.html, but the author of the question said it was "linking" errors but did not specify how to resolve them. 

Finally, I searched where the makefile was sending the compiled mex file to and, in the directory:
cd /usr/local/MATLAB/R2016b/toolbox/local
and I found the file nlopt_optimize.mexa64 which means the file has already been compiled into a mex function, yet MATLAB cannot find the function properly when I ran the test script. 

Am I overlooking something? How can I address this issue?

**Comment by jschueller** [comment]
- what's the ouput of matlab -n ? that should help you set the matlab dir
- I dont know mex, but I think you're not linking to nlopt with that command

**Comment by karenacai** [comment]
I figured out the solution. In the MATLAB editor, I first added the following paths '/usr/local/lib' and  '/usr/local/include' with the following commands:
`addpath('/usr/local/lib'); addpath('/usr/local/lib');`

The /usr/local/lib directory contains the libnlopt.so library file, and the /usr/local/include directory contains the nlopt.h header file. 

Finally, we link the library file to MATLAB via the command: 
`loadlibrary('libnlopt', 'nlopt.h')`

Now, my example MATLAB code is working properly!
